lib/system/firmware.toit

```diff
@@ -105,22 +105,71 @@ It is common that newly installed firmware boots with
   pending validation; see $is_validation_pending.
 */
 class FirmwareWriter extends ServiceResourceProxy:
+  static BUFFER_SIZE_ ::= 4096
+  buffer_/ByteArray? := null
+  buffered_/int := 0
+
   constructor from/int to/int:
     if not _client_: throw "UNSUPPORTED"
     super _client_ (_client_.firmware_writer_open from to)
 
   write bytes/ByteArray -> none:
```

Comment:
Does this function deserve a comment saying that it may neuter the byte arrays that are passed to it?

Reply:
Done.